### PR TITLE
Remove legacy `color` tokens

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHY are these changes introduced?

Part of #10661 

### WHAT is this pull request doing?

Removes all deprecated v11 `color` custom properties from Polaris.

### CODE REVIEW

File: polaris-tokens/src/themes/base/color.ts [hunk 1]
```diff
@@ -400,1106 +235,631 @@ export const color: {
...
-    value: colorsExperimental.blackAlpha[1],
+    value: colors.blackAlpha[1],
   },
   'color-bg-fill-transparent-hover': {
-    value: colorsExperimental.blackAlpha[5],
+    value: colors.blackAlpha[5],
   },
   'color-bg-fill-transparent-active': {
-    value: colorsExperimental.blackAlpha[6],
+    value: colors.blackAlpha[6],
```
Comment (lgriffee) [inline polaris-tokens/src/themes/base/color.ts]:
```suggestion
    value: colors.blackAlpha[7],
```
This is listed as [7 in Figma](https://www.figma.com/file/oKq2CLPYeX5qK5frzwipM3/Polaris-Gen-3-Styles?type=design&node-id=14696%3A1999&mode=design&t=vGyfRzAa3mogdM3j-1). We need to determine which of these values is correct.

Screenshot 2023-10-02 at 11 28 44 AM

File: polaris-tokens/src/themes/base/color.ts [hunk 2]
```diff
@@ -400,1106 +235,631 @@ export const color: {
...
-  'color-icon-interactive': {
-    value: colorsExperimental.blue[13],
-  },
-  'color-icon-interactive-inverse': {
-    value: colorsExperimental.blue[8],
+    value: colors.orange[11],
   },
   'color-icon-magic': {
-    value: colorsExperimental.purple[13],
+    value: colors.purple[13],
```
Comment (lgriffee) [inline polaris-tokens/src/themes/base/color.ts]:
```suggestion
    value: colors.purple[12],
```
This is listed as [12 in Figma](https://www.figma.com/file/oKq2CLPYeX5qK5frzwipM3/Polaris-Gen-3-Styles?type=design&node-id=14696%3A1737&mode=design&t=vGyfRzAa3mogdM3j-1). We need to determine which of these values is correct.

Screenshot 2023-10-02 at 11 56 22 AM

File: polaris-tokens/src/themes/base/color.ts [hunk 3]
```diff
@@ -400,1106 +235,631 @@ export const color: {
...
-    value: colorsExperimental.blackAlpha[8],
+    value: colors.blackAlpha[8],
   },
   'color-bg-fill-transparent-secondary-hover': {
-    value: colorsExperimental.blackAlpha[7],
+    value: colors.blackAlpha[7],
   },
   'color-bg-fill-transparent-selected': {
-    value: colorsExperimental.blackAlpha[6],
+    value: colors.blackAlpha[6],
```
Comment (lgriffee) [inline polaris-tokens/src/themes/base/color.ts]:
```suggestion
    value: colors.blackAlpha[7],
```
This is listed as [7 in Figma](https://www.figma.com/file/oKq2CLPYeX5qK5frzwipM3/Polaris-Gen-3-Styles?type=design&node-id=14696%3A1737&mode=design&t=vGyfRzAa3mogdM3j-1). We need to determine which of these values is correct.

Screenshot 2023-10-02 at 12 09 02 PM

File: polaris-tokens/src/themes/base/color.ts [hunk 4]
```diff
@@ -400,1106 +235,631 @@ export const color: {
...
   },
   'color-bg-fill-transparent': {
-    value: colorsExperimental.blackAlpha[1],
+    value: colors.blackAlpha[1],
```
Comment (lgriffee) [inline polaris-tokens/src/themes/base/color.ts]:
```suggestion
    value: colors.blackAlpha[3],
```
This is listed as [3 in Figma](https://www.figma.com/file/oKq2CLPYeX5qK5frzwipM3/Polaris-Gen-3-Styles?type=design&node-id=14696%3A1737&mode=design&t=vGyfRzAa3mogdM3j-1). We need to determine which of these values is correct.

Screenshot 2023-10-02 at 12 12 38 PM

**Comment by sarahill** [comment]
I don't see `icon-inverse-secondary` -> `gray[12]`

**Review by sarahill — APPROVED** [review]
Laura caught all the values that needed to be updated ❤️ 

Looks good with those updates 🎉

**Review by sophschneider — APPROVED** [review]
Storybook regression tests looked good!

**Review by laurkim — APPROVED** [review]

**Pull request merged**